packages/scandipwa/src/component/SearchOverlay/SearchOverlay.component.tsx

```diff
@@ -30,11 +30,19 @@ export class SearchOverlayComponent extends PureComponent<SearchOverlayComponent
         searchCriteria: '',
     };
 
+    state: SearchOverlayComponentState = {
+        prevSearchCriteria: '',
+        activeClosingAnimation: false,
+    };
+
     timeout: NodeJS.Timeout | null = null;
 
+    resultRef = createRef<HTMLDivElement>();
+
     componentDidUpdate(prevProps: SearchOverlayComponentProps): void {
         const { searchCriteria: prevSearchCriteria } = prevProps;
         const { searchCriteria, clearSearchResults, makeSearchRequest } = this.props;
+        this.setState({ prevSearchCriteria });
```

Comment:
move the `setState` call inside of the if check scope, to make sure it's only updated when necessary, otherwise it will trigger the component update every time.